File: src/main/java/com/kehsa/spring_rest/database/entities/SubGroup.java
package com.kehsa.spring_rest.database.entities;

import lombok.*;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "sub_groups")
@AllArgsConstructor
@NoArgsConstructor
@Getter @Setter
@EqualsAndHashCode
public class SubGroup {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String name;
    private boolean deleted;
    @ManyToOne
    private Group group;
    @OneToMany(mappedBy = "subGroup")
    private List<Good> goods;

    @PostLoad
    private void postLoad() {
        goods.removeIf(Good::isDeleted);
    }
}

File: webpack_src/data/SubGroup.ts
export type SubGroups = {
    _embedded: {
        groups: SubGroup[]
    }
}

export type SubGroup = {
    id?: number,
    name: string,
    deleted: boolean,
    group?: string,
    goods?: string[],
    _links: {
        self: { href: string },
        subGroup: { href: string },
        group: { href: string },
        goods: { href: string }
    }
}
